src/rpc/claimtrie.cpp

```diff
@@ -380,6 +380,10 @@ UniValue getclaimbyid(const UniValue& params, bool fHelp)
         );
 
     LOCK(cs_main);
+
+    // Used for validation.
+    ParseHashV(params[0], "Claim-id (parameter 1)");
```

Comment:
This check is probably wrong. Are IDs a hex value or numeric?

Reply:
https://github.com/lbryio/lbrycrd/blob/master/src/rpc/claimtrie.cpp#L359

ClaimIds are the hashes represented as hex strings.

Reply:
It's not wrong, but also not quite right.
ParseHashV is for hex strings representing total 256 bits for txid's.
ClaimId's are 160 bit so is represented by 40 hex chars. 
I would prefer you create a function similar to ParseHashV in rpc/claimtrie.cpp that is specifically for ClaimId's, maybe called ParseClaimId()

Reply:
Would it be acceptable to use `ParseHexV` instead of creating a new method? It does return a vector so there would be an unnecessary allocation, but it does the right thing without writing new code.

Reply:
No, I think it is not acceptable. If you enter in the wrong amount of characters, it should be able to catch that.

If you don't want to write it though, I can do it. Or maybe @tzarebczan should give you more tip for the work. Let me know.

Reply:
Ok, that makes sense. It looks like `ParseHashV` doesn't do length validation either, perhaps it should be updated as well to check that the length is 32?
```
uint256 ParseHashV(const UniValue& v, string strName)
{
    string strHex;
    if (v.isStr())
        strHex = v.get_str();
    if (!IsHex(strHex)) // Note: IsHex("") is false
        throw JSONRPCError(RPC_INVALID_PARAMETER, strName+" must be hexadecimal string (not '"+strHex+"')");
    uint256 result;
    result.SetHex(strHex);
    return result;
}
```

Reply:
On the most recent Bitcoin Core code base has updated the ParseHashV function to check this: https://github.com/bitcoin/bitcoin/blob/master/src/rpc/server.cpp#L118

So no you do not need to update ParseHashV since we plan to merge with the most recent Bitcoin Core code base soon.